lib/handler/cache-handler.js

```diff
@@ -241,10 +241,13 @@ class CacheHandler {
  * @param {import('../../types/cache-interceptor.d.ts').default.CacheControlDirectives} cacheControlDirectives
  */
 function canCacheResponse (cacheType, statusCode, resHeaders, cacheControlDirectives) {
-  // Allow caching for status codes 200 and 307 (original behavior)
-  // Also allow caching for other status codes that are heuristically cacheable
-  // when they have explicit cache directives
-  if (statusCode !== 200 && statusCode !== 307 && !HEURISTICALLY_CACHEABLE_STATUS_CODES.includes(statusCode)) {
```

Comment:
Having a heuristically cacheable status code matters only for responses without an explicit caching directive. It is not a requirement for responses with an explicit caching directive, see [Storing Responses in Caches](https://datatracker.ietf.org/doc/html/rfc9111#name-storing-responses-in-caches).

This previous code also allows to cache 307 responses without an explicit caching directive if the `cacheByDefault` option is used. But 307 is not heuristically cacheable, and so, it should not be cached if its response lacks explicit caching directive..